libsolidity/parsing/Token.cpp

```diff
@@ -50,6 +50,67 @@ namespace dev
 namespace solidity
 {
 
+void ElementaryTypeNameToken::parseDetails(Token::Value _baseType, unsigned const& _first, unsigned const& _second)
+{
+	solAssert(Token::isElementaryTypeName(_baseType), "");
+	string baseType = Token::toString(_baseType);
+	if (_first == 0 && _second == 0)
+	{
+		m_name = baseType;
+		m_firstNumber = _first;
+		m_secondNumber = _second;
+	}
+
+	if (baseType == "bytesM")
+	{
+		solAssert(_second == 0, "There should not be a second size argument to type bytesM.");
+		for (unsigned m = 1; m <= 32; m++)
```

Comment:
This was off of a previous suggestion. I think this could be optimized, yes. Perhaps a SolAssert.